src/libraries/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Unix.tt

```diff
@@ -0,0 +1,89 @@
+﻿<#@ template debug="false" hostspecific="false" language="C#" #>
+<#@ import namespace="System" #>
+<#@ assembly name="System.Core" #>
+<#@ assembly name="System.Net.Primitives" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ output extension=".cs" #>
+<#@ include file="TlsCipherSuiteNameParser.ttinclude" #><#@
+    include file="TlsCipherSuite.cs" #>
+<# Array enumValues = typeof(TlsCipherSuite).GetEnumValues(); #>
+
+
+using System.Diagnostics;
+using System.Security.Authentication;
+
+namespace System.Net.Security
+{
+    internal partial struct SslConnectionInfo
+    {
+        private void MapCipherSuite(TlsCipherSuite cipherSuite)
+        {
+            TlsCipherSuite = cipherSuite;
+            KeyExchKeySize = 0;
+
+            int index = (int)cipherSuite * 2;
+            Debug.Assert(s_cipherToIndex.IndexOf(index) > -1, $"No mapping found for cipherSuite {cipherSuite}");
+            ulong data = s_encodedNumber[s_cipherToIndex[index + 1]];
```

Comment:
`TlsCipherSuite` members are not consecutive integers. There are large gaps. Indexing like this won't work. I think that the first thing that any solution should do is to switch on `cipherSuite` that makes Roslyn to generate efficient switching code for us.

Reply:
Ah yes. `TlsCipherSuite` is generated by another T4 template. I copied its guts into this .tt file and created a jump map. That increased both the size of NativeAOT app and the assembly. I will revert this version back to 09cde7d2aad761c648f5b1c79f08ff285158fd27. 🥲